test/kitchen/drivers/azure-driver.yml

```diff
@@ -17,7 +17,8 @@
 provisioner:
   name: chef_solo
   product_name: chef
-  product_version: 14.12.9
+  product_version: 15.16.4
```

Comment:
Out of curiosity, was chef bumped because the older version was missing support for ARM?

Reply:
Yes. @KSerrania was the one who found this out, so he might have more details.

Reply:
Indeed, arm64 support was added in Chef 15 (Chef 14 had arm64 support for CentOS only).

However, setting that for the Azure driver means that all our kitchen tests are going to run with Chef 15. Have we checked that they work? I'm afraid that some of the older platforms we run the regular kitchen tests on (Ubuntu 14.04, Debian 8) may not be supported by Chef 15.

Reply:
Are you talking about non-CI kitchen tests? I had to upgrade it for both because I needed to check for architecture in our recipe.

Reply:
I'm talking about the kitchen tests we run on Azure in the CI, in the `kitchen_testing` stage.

Reply:
As I feared, kitchen tests fail on Debian 8 and Ubuntu 14.04 because they're not supported by Chef 15.

Moreover, for a yet unknown reason, all Windows tests (on all platforms) fail when installing the tools needed to run the tests:
```
Installing Busser plugins: busser-rspec

busser.bat : 'ruby.exe' is not recognized as an internal or external command,
At line:31 char:2
+ & <<<<  "$busser" plugin install $plugins.Split() 2>&1
    + CategoryInfo          : NotSpecified: ('ruby.exe' is n...ternal command,:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
operable program or batch file.
```

If the only reason you need Chef 15 is because of the `arm?` helper, you could go back to Chef 14 (while we find a path to upgrade Chef) and define that helper directly in your recipe: https://github.com/DataDog/datadog-agent/commit/2117d9a044272252ae8dc5c9596a72371dffa5d0, https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/4297554

Reply:
Using a `CHEF_VERSION` override seems to have fixed it: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/4300973